pages/api/applications/index.ts

```diff
@@ -240,4 +216,30 @@ function validatePostRequest(req: NextApiRequest): Nullable<MessageType> {
   return null;
 }
 
+function convertApplicationListDataToPayload({
+  id,
+  role,
+  latestStage,
+  taskNotificationCount,
+}: {
+  id: number;
+  role: Role & { company: Company };
+  latestStage: ApplicationStage;
+  taskNotificationCount: number;
+}): ApplicationListData {
+  const { company, title, type, year } = role;
+  return {
+    id,
+    taskNotificationCount,
+    role: {
+      type,
+      title,
+      year,
+      id: role.id,
+      company: { id: company.id, name: company.name, companyUrl: company.companyUrl },
+    },
+    latestStage: convertApplicationStageToPayload(latestStage),
```

Comment:
`latestStage` may be undefined if an application has no stages.
<img width="676" alt="Screenshot 2022-09-16 at 12 29 35 PM" src="https://user-images.githubusercontent.com/71375383/190557030-a31232b5-bbcf-45bf-af50-23aee40218aa.png">